src/pages/Flavors.js

```diff
@@ -1,36 +1,198 @@
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
 import { Helmet } from 'react-helmet';
 import React, { Component } from 'react';
-import { Container, Table } from 'react-bootstrap';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { PagerInfo, withPagination } from 'components/Pagination/Pagination';
+import { Col, Container, Form, Row, Table } from 'react-bootstrap';
+import ToggleButton from 'components/ToggleButton/ToggleButton';
+import { actions as appActions } from 'reducers/application';
+import { actions as flavorActions } from 'reducers/flavor';
+import { actions as flavorsActions } from 'reducers/flavors';
+import { isLoggedIn } from 'selectors/application';
+import { getStash, isLoaded } from 'selectors/flavor';
+import { getAllFlavors, getFlavorsPager } from 'selectors/flavors';
 
-import FlavorData from 'data/flavors';
+export class Flavors extends Component {
+  static propTypes = {
+    actions: PropTypes.object.isRequired,
+    collection: PropTypes.array.isRequired,
+    loggedIn: PropTypes.bool.isRequired,
+    pager: PropTypes.object.isRequired,
+    pagerNavigation: PropTypes.node.isRequired,
+    stash: PropTypes.array,
+    stashLoaded: PropTypes.bool.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = { stashControl: false };
+
+    const { actions, loggedIn, stashLoaded } = this.props;
+
+    if (loggedIn) {
+      if (!stashLoaded) {
+        actions.requestStash();
+      }
+      this.handleStashControl = this.stashController.bind(this);
+      this.handleAddToStash = this.addToStash.bind(this);
+      this.handleRemoveFromStash = this.removeFromStash.bind(this);
+    }
+  }
+
+  stashController(event) {
+    const { stash } = this.props;
+    const { holdings: stashMap } = this.state;
+
+    if (!stashMap) {
+      const holdings = [];
+
+      stash.map(flavor => {
+        holdings[flavor.flavorId] = true;
+      });
+      this.setState({ holdings });
+    }
+    const {
+      target: { checked, name }
+    } = event;
+
+    this.setState({
+      [name]: checked
+    });
+  }
+
+  addToStash(id) {
+    const { actions } = this.props;
+
+    actions.addStash({ id });
+
+    const { holdings } = this.state;
+
+    holdings[id] = true;
+
+    this.setState({ holdings });
+  }
+
+  removeFromStash(id) {
+    const { actions } = this.props;
+
+    actions.removeStash({ id });
+
+    const { holdings } = this.state;
+
+    holdings[id] = false;
+
+    this.setState({ holdings });
+  }
+
+  inStashIcon(id) {
```

Comment:
inStashIcon and noStashIcon should be the same getter - `onClick` would then have an `if { } else { }` or ternary construct that invokes the proper method based on the component state. We should change ToggleButton to pass the current state as a boolean instead of `event` to the `onClick` handler here, that will make this much easier.

Reply:
Merged